AsyncDisplayKit/ASControlNode.mm

```diff
@@ -316,13 +329,28 @@ - (NSArray *)actionsForTarget:(id)target forControlEvent:(ASControlNodeEvent)con
 
   ASDN::MutexLocker l(_controlLock);
   
-  // Grab the event dispatch table for this event.
-  NSMapTable *eventDispatchTable = _controlEventDispatchTable[_ASControlNodeEventKeyForControlEvent(controlEvent)];
-  if (!eventDispatchTable)
+  // Grab the event target action array for this event.
+  NSMutableOrderedSet *eventTargetActionArray = _controlEventDispatchTable[_ASControlNodeEventKeyForControlEvent(controlEvent)];
+  if (!eventTargetActionArray) {
     return nil;
+  }
 
-  // Return the actions for this target.
-  return [eventDispatchTable objectForKey:target];
+  NSMutableArray *actions = [[NSMutableArray alloc] init];
+  
+  // Collect all actions for this target.
+  for (ASControlTargetAction *targetAction in eventTargetActionArray) {
+    if (target && targetAction.target && target != targetAction.target) {
+      continue;
+    }
+    
+    if (!target && !targetAction.createdWithNoTarget) {
+      continue;
+    }
+    
+    [actions addObject:NSStringFromSelector(targetAction.action)];
```

Comment:
This logic is a little tough for my pea-brain to handle. Would this logic be equivalent:

```objectivec
if ((target == nil && targetAction.createdWithNoTarget) || (target != nil && target == targetAction.target)) {
  [actions addObject:NSStringFromSelector(targetAction.action)];
}
```